peercrawler.py

```diff
@@ -325,6 +329,39 @@ def __str__(self):
 
         return s
 
+    def serialize(self) -> str:
+        graph_copy = self.get_connections_graph()
+        nodes = {}
+        for peer, connections in graph_copy.items():
+            peer_data = vars(peer)
+            peer_data["connections"] = [str(id(c)) for c in connections]
+            nodes[id(peer)] = peer_data
+
+        json_connections = json.dumps(nodes, cls=jsonencoder.NanoJSONEncoder)
+        return json_connections
+
+    @staticmethod
+    def deserialize(data: str) -> dict[Peer, peer_set]:
```

Comment:
It would be nice to be able to deserialise form a json object, not just a json string.